diff --git a/verus-mimalloc/os_mem.rs b/verus-mimalloc/os_mem.rs
--- a/verus-mimalloc/os_mem.rs
+++ b/verus-mimalloc/os_mem.rs
@@ -131,6 +131,7 @@
         pt.0.addr() != MAP_FAILED ==> pt.1@.os_has_range_no_read_write(pt.0 as int, len as int),
         pt.0.addr() != MAP_FAILED ==> pt.0.addr() + len < usize::MAX,
+        pt.0.addr() != MAP_FAILED ==> pt.0 as int % page_size() == 0,
         pt.0.addr() != MAP_FAILED ==> pt.0@.provenance == pt.1@.points_to.provenance(),
 {
